source/auth/auth.rst

```diff
@@ -87,6 +87,11 @@ mechanism (string)
 mechanism_properties
 	* Includes additional properties for the given mechanism.
 
+Specifying userinfo
```

Comment:
I wonder if this reads more clearly:

> *User information in URI implies authentication*
> 
> The presence of a credential delimiter (i.e. `@`) in the URI connection string is evidence that the user has unambiguously specified user information and MUST be interpreted as a user configuring authentication credentials (even if the username and/or password are empty strings).

This mostly keeps the first half of your original text but modifies the second half to borrow language from the subsequent section ("Ambiguity between authentication source and URI database"). Since we're talking about requiring driver behavior, it's helpful to use actionable terms (e.g. MUST, MUST NOT).

Alternative title: "Credential delimiter in URI implies authentication"

Reply:
Yes that makes sense to me!